server/app/mutations/grid_certificates/remove_certificate.rb

```diff
@@ -0,0 +1,23 @@
+
+require_relative 'common'
+require_relative '../../services/logging'
+
+module GridCertificates
+  class RemoveCertificate < Mutations::Command
+    include Common
+    include Logging
+
+    required do
+      model :certificate, class: Certificate
+    end
+
+    def validate
+      services_using = certificate.grid.grid_services.where(:'certificates.subject' => certificate.subject).map { |s| s.to_path}
+      add_error(self.certificate.subject, :certificate_in_use, "Certificate still in use in services: #{services_using.join(',')}") if services_using.size > 0
```

Comment:
Bad idea to use the actual subject as the error key, it should be something constant like `:certificate` instead.

```
 [fail] Removing certificate for test.188.226.131.23.xip.io from development grid      
 [error] 422 : Unprocessable Entity:
	test:
	  '188':
	    '226':
	      '131':
	        '23':
	          xip:
	            io: 'Certificate still in use in services: development/ingress-lb/lb'
```

Reply:
oh, totally forgot the dotted functionality there.

Reply:
Replace `add_error(self.certificate.subject, ...)` with just `add_error(:certificate, ...)`